added tf state backend in azure

vcosqui opened the pull request.

Commit:
added tf state backend in azure

ogomezso commented on azure/providers.tf:
```diff
@@ -0,0 +1,9 @@
+      resource_group_name  = "migrations"
+      storage_account_name = "migrationstfstateazure"
+      container_name       = "tfstate"
+      key                  = "terraform.tfstate"
```
I would extract these setting to tf variables

ogomezso replied:
And perhaps all the backend configuration to a backend.tf file so anyone can easily switch to local backend just avoiding this file

ogomezso reviewed (changes requested):
I would extract the blob storage settings to tf variables

vcosqui commented:
I am afraid it can't be done
apparently init phase does not have access to variables
https://stackoverflow.com/a/65010204

ogomezso commented:
> I am afraid it can't be done apparently init phase does not have access to variables https://stackoverflow.com/a/65010204

So perhaps we can put it as plain env variables or even placeholders, I'm trying to avoid having references to specific resources on this repo.

Thanks!

Commit:
added tf state backend in azure with external backend settings, run with

```tf init -backend-config "resource_group_name=migrations" \
 -backend-config "storage_account_name=migrationstfstateazure" \
 -backend-config "container_name=tfstate" \
 -backend-config "key=terraform.tfstate"```

Commit:
add owner email tag
